src/pykeen/nn/node_piece/tokenization.py

```diff
@@ -150,6 +151,91 @@ def __call__(
+        return self._call(
+            edge_index=get_edge_index(mapped_triples=mapped_triples),
+            num_tokens=num_tokens,
+            num_entities=num_entities,
+        )
+
+
+class MetisAnchorTokenizer(AnchorTokenizer):
+    """
+    An anchor tokenizer, which first partitions the graph using Metis.
+
+    We use the binding by `torch-sparse`. The Metis graph partitioning algorithm is described here:
+    http://glaros.dtc.umn.edu/gkhome/metis/metis/overview
+    """
+
+    def __init__(self, num_partitions: int = 2, **kwargs):
+        """Initialize the tokenizer.
+
+        :param num_partitions:
+            the number of partitions obtained through Metis.
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`AnchorTokenizer.__init__`. note that there will be one
+            anchor tokenizer per partition, i.e., the vocabulary size will grow respectively.
+        """
+        super().__init__(**kwargs)
+        self.num_partitions = num_partitions
+
+    # docstr-coverage: inherited
+    def __call__(
+        self,
+        mapped_triples: MappedTriples,
+        num_tokens: int,
+        num_entities: int,
+        num_relations: int,
+    ) -> Tuple[int, torch.LongTensor]:  # noqa: D102
+        try:
+            import torch_sparse
+        except ImportError as err:
+            raise ImportError(f"{self.__class__.__name__} requires `torch_sparse` to be installed.") from err
+
+        logger.info(f"Partitioning the graph into {self.num_partitions} partitions.")
+        row, col = get_edge_index(mapped_triples=mapped_triples)
+        re_ordered_adjacency, bound, perm = torch_sparse.partition(
+            src=torch_sparse.SparseTensor(row=row, col=col, sparse_sizes=(num_entities, num_entities)),
+            num_parts=self.num_partitions,
+            recursive=True,
+        )
+        sizes = bound.diff()
+        logger.info(f"Partition sizes: min: {sizes.min().item()}, max:{sizes.max().item()}")
+
+        # select independently per partition
+        vocabulary_size = 0
+        assignment = []
+        edge_count = 0
+        for low, high in more_itertools.pairwise(bound.tolist()):
+            # select adjacency part;
+            # note: the indices will automatically be in [0, ..., high - low), since they are *local* indices
+            edge_index = re_ordered_adjacency[low:high, low:high].to_torch_sparse_coo_tensor().coalesce().indices()
+            edge_count += edge_index.shape[1]
+            num_entities = high - low
+            this_vocabulary_size, this_assignment = super(self.__class__, self)._call(
+                edge_index=edge_index, num_tokens=num_tokens, num_entities=num_entities
+            )
+            assert this_assignment.shape[0] == num_entities
+            # offset
+            this_assignment = this_assignment + vocabulary_size
+            # note: permutation will be later on reverted
+            vocabulary_size += this_vocabulary_size
```

Comment:
we have a few problems with the padding tokens here:
* the -1 token exists only in the very first assignment, then they get re-written
* the total vocab size gets to be `(anchors_per_partition + 1) * num_partitions` whereas the padding token should be shared for all of them, s.t. the total vocab size is still `anchors_per_partition * num_partitions + 1`